contracts/oracle/GOhmEthOracle.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
+import {Decimal, IOracle} from "./IOracle.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+
+interface IgOHM {
+    function mint(address _to, uint256 _amount) external;
+
+    function burn(address _from, uint256 _amount) external;
+
+    function index() external view returns (uint256);
+
+    function balanceFrom(uint256 _amount) external view returns (uint256);
+
+    function balanceTo(uint256 _amount) external view returns (uint256);
+
+    function migrate(address _staking, address _sOHM) external;
+}
+
+/// @title GOHM Eth Oracle
+/// @notice Report the gOHM price in units of ETH. Calculated by reading a Chainlink OHM V2
+///         oracle price and multiplying that by the queried OHM index
+contract GOhmEthOracle is IOracle, CoreRef {
+    using Decimal for Decimal.D256;
+
+    /// @notice the referenced chainlink oracle
+    AggregatorV3Interface public immutable chainlinkOHMETHOracle;
+
+    /// @notice Oracle decimals normalizer
+    uint256 public oracleDecimalsNormalizer;
+
+    /// @notice gOHM token address.
+    IgOHM public constant GOHM = IgOHM(0x0ab87046fBb341D058F17CBC4c1133F25a20a52f);
+
+    /// @notice Maximum time since last update that the Chainlink oracle is considered valid
+    uint256 public constant MAX_ORACLE_UPDATE_TIME = 86400 seconds;
```

Comment:
Sourced from the Chainlink data feed: https://data.chain.link/ethereum/mainnet/crypto-eth/ohmv2-eth